File: Leetcode/Easy Problems/1342. Number of Steps to Reduce a Number to Zero/dec-19-2023.py
#  December 19, 2023
#  - Runtime Achieved: 43 ms
#  - Memory Achieved: 16.2 MB
class Solution:
    def numberOfSteps(self, num: int) -> int:
        count = 0
        while num > 0:
            count += 1
            if num % 2 == 0:
                num /= 2
            else:
                num -= 1
        return count

File: Leetcode/Easy Problems/383. Ransom Note/dec-19-2023.cpp
/*
  December 19, 2023
  - Runtime Achieved: 12 ms
  - Memory Achieved: 9.8 MB
*/
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char,int> chart;
        bool result = true;
        string a = ransomNote;
        string b = magazine;

        for(char i:b) {
            chart[i]++;
        }
        for(char j:a) {
            if(chart[j] == 0)
                result = false;
            chart[j]--;  
        }
        return result;
    }
};
